source/iterator/combined.go

```diff
@@ -107,7 +107,30 @@ func (c *Combined) HasNext(ctx context.Context) (bool, error) {
 func (c *Combined) Next(ctx context.Context) (sdk.Record, error) {
 	switch {
 	case c.snapshot != nil:
-		return c.snapshot.Next(ctx)
+		record, err := c.snapshot.Next(ctx)
+		if err != nil {
+			return sdk.Record{}, fmt.Errorf("snapshot next: %w", err)
+		}
+
+		hasNext, err := c.snapshot.HasNext(ctx)
+		if err != nil {
+			return sdk.Record{}, fmt.Errorf("snapshot has next: %w", err)
+		}
+
+		if !hasNext {
+			sdk.Logger(ctx).Debug().Msgf("switching to the CDC mode")
+
+			if err = c.switchToCDCIterator(ctx); err != nil {
```

Comment:
Do we have some reason why we have to do it here and not do it inside hasNext method?

Reply:
Okay, let me explain this behavior a bit better.

For example, we created a pipeline with the HubSpot source connector and some destination. We set the `pollingPeriod` value to 5 seconds and the `bufferSize` to 2. In HubSpot, we had 4 items. We started the pipeline, and right after that two items had been inserted into a destination. We waited for the next two items for 5 seconds. 5 seconds passed and the items were inserted. There were no items left so the connector switched to the CDC mode but the last position's mode was Snapshot and it hadn't changed. And then we decided to stop the pipeline. While the pipeline was stopped we updated an item. We restarted the pipeline and don't see the updated item. That's because after the restart the position's mode was still Snapshot so the connector first created a Snapshot iterator and then switched to the CDC listening for items that are updated after the current date and didn't see that item that we updated when the pipeline was stopped.

Reply:
I've found the solution to avoid this as well as the `ConvertToCDCPosition` from the comment above.

Reply:
In this commit: https://github.com/conduitio-labs/conduit-connector-hubspot/pull/14/commits/56283f003bd28898f2adf436daef035a554d24a1